rbd.c

```diff
@@ -943,12 +943,21 @@ static void rbd_finish_aio_generic(rbd_completion_t completion,
 {
 	struct tcmu_device *dev = aio_cb->dev;
 	struct tcmulib_cmd *tcmulib_cmd = aio_cb->tcmulib_cmd;
-	struct iovec *iovec = tcmulib_cmd->iovec;
-	size_t iov_cnt = tcmulib_cmd->iov_cnt;
 	uint32_t cmp_offset;
 	uint16_t asc_ascq;
 	int64_t ret;
 	int tcmu_r;
+	struct iovec *iovec;
+	size_t iov_cnt;
+
+	if ((tcmulib_cmd->cdb[0] & 0x83) == 0x83) {
```

Comment:
As I said before the handlers should not be digging into the low level details. I had also said we should start to rewrite this code. However, let's do a quick fix here first. It looks like a rbd.c specific bug that does the read/write callouts but also digs into the tcmulib_cmd. It should just be using the iovec passed into the read/write callouts.

So for now just add new fields onto rbd_aio_cb  for the iovec and iov_cnt. Have tcmu_rbd_write/read set the new fields to the values passed in. In the completion function then just access that, so rbd.c is not digging into the tcmulib_cmd fields.

Reply:
Oh yeah, when you re-upload this next time break the patches in to 2 patches. One for the xcopy issue and one for the CAW seek issue. Thanks.

Reply:
Hey, @umuzhaohui  I saw you uploaded new versions of these patches. Were you also going to handle this review comment?

Here is a untested patch that does what I was requesting. Does that fix your issue?

[rbd-dont-access-cmd-iov.txt](https://github.com/open-iscsi/tcmu-runner/files/1790624/rbd-dont-access-cmd-iov.txt)